WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanViewModel.kt

```diff
@@ -90,19 +90,22 @@ class ScanViewModel @Inject constructor(
     }
 
     private fun init() {
+        updateUiState(FullScreenLoadingUiState)
         launch {
             scanStateModel = scanStore.getScanStateForSite(this@ScanViewModel.site)
             scanStateModel?.let {
-                updateUiState(buildContentUiState(it))
                 if (fixableThreatIds.isNotEmpty()) fetchFixThreatsStatus(fixableThreatIds, isInvokedByUser = false)
             }
-            fetchScanState()
+            fetchScanState(isInvokedFromInit = true)
         }
     }
 
-    private fun fetchScanState(invokedByUser: Boolean = false, isRetry: Boolean = false) {
+    private fun fetchScanState(
```

Comment:
Suggestion (💡): This suggestion is not based on the changes you did on this PR, but rather a general suggestion to improve on `fetchScanState(...)` signature and how it is being used. So, here I go, what if we update the signature and make this function a `suspend` function in order to avoid the `launch { ... }` being called twice on most of cases?

Reply:
Nice finding 🙌, signature changed in a37f45a.

Reply:
Thanks for the change @ashiagr ! 🥇